relay-general/src/pii/processor.rs

```diff
@@ -879,3 +915,38 @@ fn test_logentry_value_types() {
             .is_none());
     }
 }
+
+#[test]
+fn test_ip_address_hashing() {
+    let config = PiiConfig::from_json(
+        r##"
+            {
+                "applications": {
+                    "$user.ip_address": ["@ip:hash"]
+                }
+            }
+        "##,
+    )
+    .unwrap();
+
+    let mut event = Annotated::new(Event {
+        user: Annotated::new(User {
+            ip_address: Annotated::new(IpAddr("127.0.0.1".to_string())),
+            ..Default::default()
+        }),
+        ..Default::default()
+    });
+
+    let compiled = config.compiled();
+    let mut processor = PiiProcessor::new(&compiled);
+    process_value(&mut event, &mut processor, ProcessingState::root()).unwrap();
+
+    let user = event.value().unwrap().user.value().unwrap();
+
+    assert!(user.ip_address.value().is_none());
+
+    assert_eq!(
+        user.id.value().unwrap().as_str(),
+        "AE12FE3B5F129B5CC4CDD2B136B7B7947C4D2741"
+    );
+}
```

Comment:
Could you please add a second test for the other case that this PR is supposed to solve: A user with an explicit ID that does not get overwritten.

Reply:
done